extensions/guacamole-auth-restrict/src/main/resources/controllers/hostRestrictionFieldController.js

```diff
@@ -0,0 +1,168 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+
+/**
+ * Controller for host restriction fields, which are used to configure a
+ * hostname, IP address, or CIDR range, that this restriction applies to.
+ */
+angular.module('guacRestrict').controller('hostRestrictionFieldController', ['$scope', '$injector',
+    function hostRestrictionFieldController($scope, $injector) {
+       
+    // Required types
+    const HostRestrictionEntry = $injector.get('HostRestrictionEntry');
+        
+    /**
+     * Options which dictate the behavior of the input field model, as defined
+     * by https://docs.angularjs.org/api/ng/directive/ngModelOptions
+     *
+     * @type Object.<String, String>
+     */
+    $scope.modelOptions = {
+        
+        /**
+         * Space-delimited list of events on which the model will be updated.
+         *
+         * @type String
+         */
+        updateOn : 'blur',
+
+        /**
+         * The time zone to use when reading/writing the Date object of the
+         * model.
+         *
+         * @type String
+         */
+        timezone : 'UTC'
+
+    };
+    
+    /**
+     * The restrictions, as objects, that are used by the HTML template to
+     * present the restrictions to the user via the web interface.
```

Comment:
This array should be annotated with the `@type` of item that it holds.

Reply:
Fixed via rebase.